### PR TITLE
Count users that read another page before Omniture fires

### PR DESCRIPTION
**Pull request by gklopper**
Hypothesis - some users click a link and view the next page before Omniture has had a chance to run.

This should test that hypothesis.

### CODE REVIEW

File: common/app/views/fragments/analytics.scala.html [hunk 1]
```diff
@@ -61,3 +62,27 @@
 </noscript>
 
 <img src="@Configuration.debug.beaconUrl/count/pv.gif" alt="" style="display : none ;" rel="nofollow"/>
+
+@if(NoBounceIndicator.isSwitchedOn) {
+    @* NOTE for removal - there is a corresponding entry in omniture.js *@
+    <script type="text/javascript">
```
Comment (gklopper) [inline common/app/views/fragments/analytics.scala.html]: Inline script not ideal, but I want this separate from our Javascript system and switch will enforce deletion in the future.

File: common/app/conf/switches.scala [hunk 2]
```diff
@@ -640,7 +640,7 @@ object Switches {
     "international-edition",
     "International edition A/B test on",
     safeState = Off,
-    sellByDate = new LocalDate(2015, 7, 31),
+    sellByDate = new LocalDate(2015, 9, 30),
```
Comment (gklopper) [inline common/app/conf/switches.scala]: While I am in here, this has been extended

Comment (johnduffell) [inline common/app/conf/switches.scala]: setting switches to a random day in the month is better than a special date as we always get a flurry of switches expiring at the end of the month

**Comment by kelvin-chappell** [comment]
:+1:

File: common/app/views/fragments/analytics.scala.html [hunk 3]
```diff
@@ -61,3 +62,27 @@
...
+            @*
+                If this key is still present then the user went to a new page before Omniture ran
+                and we did not count the page view
+            *@
+            if (session.getItem('gu-bounce-test') === 'true') {
+                var img = new Image();
+                img.src = '@{Configuration.debug.beaconUrl}/count/bounce-test-present.gif';
+            }
+
+            window.sessionStorage.setItem('gu-bounce-test', 'true');
```
Comment (johnduffell) [inline common/app/views/fragments/analytics.scala.html]: did you intend to use the "session" variable declared on line 74?

File: diagnostics/app/model/diagnostics/analytics/Metric.scala [hunk 4]
```diff
@@ -13,11 +13,15 @@ object Metric extends Logging {
...
 
+  // TODO delete bounce-test-present when this goes
+  import conf.Switches.NoBounceIndicator
+
   lazy val metrics = Map(
 
     // page views
     ("pv", CountMetric("kpis-page-views")),            // raw page views - simple <img> in body, no javascript involved
     ("pva", CountMetric("kpis-analytics-page-views")), // page view fires after analytics
+    ("bounce-test-present", CountMetric("bounce-test-present")),
```
Comment (johnduffell) [inline diagnostics/app/model/diagnostics/analytics/Metric.scala]: I'd have called it user-bounced-prematurely or something, as the bounce test is always present when the switch is on so the name is confusing

Comment (gklopper) [inline diagnostics/app/model/diagnostics/analytics/Metric.scala]: Good call

File: common/app/conf/switches.scala (hunk 2 above)
Comment (gklopper) [inline common/app/conf/switches.scala]: Not my switches :smile: I keep them under control

**Comment by OliverJAsh** [comment]
Can you share the results of this? How many page views do you think we're not counting?

**Pull request merged**